enhancements/installer/single-node-installation.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Single node installation
+authors:
+  - "@eranco74"
+reviewers:
+  - "@romfreiman"
+  - "@hardys"
+  - "@crawford"
+  - "@markmc"
+approvers:
+  - "@hardys"
+  - "@crawford"
+  - "@markmc"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "https://github.com/openshift/enhancements/pull/302"
+  - "https://github.com/openshift/installer/pull/3978"
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+
+I have tried https://github.com/openshift/enhancements/pull/302
+
+
+# Single node installation
+
+Lunch “all-in-one” control plane using static pods packed in Ignition file.
+This enhancement can be the first step of what is described in https://github.com/openshift/enhancements/pull/302
+The plan is to create bootstrap like control plane using static pods for:
+etcd, kube-apiserver, kube-controller-manager & kube-scheduler.
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+Enable single node kubernetes cluster installation using static pods via Ignition.
+Add a new command to the openshift installer `create aio-ignition` that will get as input the install-config and generate aio.ign
+The aio.ign content will be similar to what we have in the bootstrap Ignition file with some changes around bootkube and kubelet services:
+- Bootkube will finish when it sets up the control plane without applying all the openshift manifests.
+- kubelet will be configured to be a part of the cluster.
+When booting an rhcos image with the aio.ign we will get a similar flow as we get in the bootstrap node and end up with basic control plane.
+   
+## Motivation
+
+Demonstrate a prototype of creating a simple static Ignition file that boots an RHCOS machine and launches a basic Kube control plane
+
+### Goals
```

Comment:
I think goals need to enumerate the components we want running. I'll throw one possible starting point out:
1. etcd
2. kube-apiserver
3. kube-controller-manager
4. kube-scheduler
5. oauth-apiserver
6. oauth-server
7. olm
8. nothing else.

This gives a kube control plane.

Reply:
@eranco74 lets list what is provided by bootstrap static pods.

Reply:
On the bootstrap node we have these static pods yamls:
1. etcd-member-pod.yaml
2. kube-apiserver-pod.yaml
3.  kube-controller-manager-pod.yaml
4. kube-scheduler-pod.yaml
5. bootstrap-pod.yaml (cluster version operator)
6. recycler-pod.yaml (doesn't seem relevant)
Running containers:
```
crictl ps | awk '{print $7}'
POD
kube-apiserver-insecure-readyz
kube-apiserver
kube-controller-manager
kube-scheduler
cluster-version-operator
etcd-metrics
etcd-member
```
Pods that show up with kubectl:
```
kubectl --kubeconfig auth/kubeconfig get pods -A
NAMESPACE     NAME                                        READY   STATUS    RESTARTS   AGE
kube-system   bootstrap-kube-apiserver-master1            2/2     Running   0          37m
kube-system   bootstrap-kube-controller-manager-master1   1/1     Running   0          37m
kube-system   bootstrap-kube-scheduler-master1            1/1     Running   0          37m
```

We also have machineconfigoperator-bootstrap-pod.yaml that runs machine-config-server (get removed once the bootstrap manage to apply all the manifests https://github.com/openshift/installer/blob/master/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L371)

Reply:
These are the static pods manifests we have in (baremetal) Openshift  master node (3 nodes installation):
```
1. etcd-pod.yaml
1. kube-apiserver-pod.yaml
2. kube-controller-manager-pod.yaml
3. kube-scheduler-pod.yaml
4. coredns.yaml 
5. haproxy.yaml
6. keepalived.yaml
7. mdns-publisher.yaml
8. recycler-pod.yaml
```

Reply:
the keeplived/coredns/haproxy are there because u're looking into BM platfrom cluster. I guess when we try similar with none, static pods will be aligned.